src/mem_mng.c

```diff
@@ -13,16 +13,12 @@ void*
 manager_register_mem_alloc(struct Compartment* comp, size_t mem_size)
 {
     // TODO better algorithm to find blocks of memory available for mapping
-    void* new_mem = mmap((void*) (comp->scratch_mem_base + comp->scratch_mem_alloc),
-                         mem_size,
-                         PROT_READ | PROT_WRITE,
-                         MAP_PRIVATE | MAP_ANONYMOUS,
-                         -1, 0);
-
+    void* new_mem = (void*) (comp->scratch_mem_base + comp->scratch_mem_alloc);
     struct mem_alloc* new_alloc = malloc(sizeof(struct mem_alloc));
     new_alloc->ptr = (uintptr_t) new_mem;
     new_alloc->size = mem_size;
     manager_insert_new_alloc(comp, new_alloc);
+    comp->scratch_mem_alloc += mem_size;
```

Comment:
Don't we need to align `mem_size`?

Reply:
What I mean is: `malloc` should return aligned pointers (this is guaranteed in, at least, OpenBSD's man page; but I'd be astonished if any `malloc` doesn't respect this). I think that if someone asks for (say) 3 bytes, the next pointer `malloc` returns won't be aligned. That can cause all sorts of odd problems, from slow-downs (cross-word reads/writes are often slow) to outright bugs (atomic read/writes no longer make sense).

Put another way: we should fix this :)

Reply:
Does the change in https://github.com/capablevms/CHERI-ELF-comp/pull/3/commits/e3b6ef61285589f1cc424ed6fdba8643f80148cb make sense?

Reply:
I would probably do `sizeof(void *)` rather than `8` (constants like this are a nightmare waiting to bite our future selves!), but otherwise yes.

Reply:
Good point; updated in https://github.com/capablevms/CHERI-ELF-comp/pull/3/commits/1c7506f1bd4e0c08b6e4b1f0491a53cc614a0c69.